lib/index.js

```diff
@@ -1723,38 +1749,33 @@ exposeMethod('pluck');
  */
 
 Stream.prototype.pickBy = function (f) {
-
-    return this.consume(function (err, x, push, next) {
+    return this.map(objectOnly(function (x) {
         var out = {};
-        if (err) {
-            push(err);
-            next();
+        var obj = x;  // variable used to transverse prototype chain
+        function testAndAdd (prop) {
+            if (!(prop in out) && f(prop, x[prop])) {
```

Comment:
Can you also add a test for the case I mentioned above? Call it "pickBy - overridden properties" or something.